umap/static/umap/js/umap.layer.js

```diff
@@ -1540,6 +1590,18 @@ U.DataLayer = L.Evented.extend({
     return this._layers[id]
   },
 
+  // TODO Add an index
+  // For now, iterate on all the features.
+  getFeatureById: function (id) {
+    console.log('looking for feature with id ', id)
+    for (const i in this._layers) {
```

Comment:
```suggestion
    for (const feature of this._layers) {
```

Would it work? Even shorter: `return this._layers.map((feature) => feature.id === id)`

Reply:
I think I tried without luck. I'll give it another go.

Reply:
🥁 … And the answer was:

```js
Object.values(this._layers).find((feature) => feature.id === id)
```

🙌